lib/src/errors.rs

```diff
@@ -17,6 +17,8 @@ pub enum Error {
     Sled { inner: SledError },
     #[fail(display = "UUID already taken")]
     UuidTaken,
+    #[fail(display = "Invalid vertex")]
+    VertexInvalid,
```

Comment:
This should be a separate error in the bin crate, since it's not produced in lib ever. Thankfully, because `main` returns `Box<dyn Error>`, different types of errors can be returned.

Reply:
Correct me if wrong here but I think this actually would be useful in lib, because if `create_edge` returns a `false` from a non-CLI call, it would still be useful error to throw. (maybe pave the way to return an error instead of a boolean value from this method in the future)

Reply:
At the library level, it's not considered an error to insert a redundant vertex. This has a couple of distinct advantages:

1. For datastores that provide all-or-none transaction semantics, inserting a vertex redundantly won't invalidate the entire transaction.
2. Similarly, for bulk inserts, this ensures that an attempt to create a vertex won't invalidate the entire bulk insert operation.